internal/gatewayapi/backendtrafficpolicy.go

```diff
@@ -125,15 +136,24 @@ func (t *Translator) ProcessBackendTrafficPolicies(backendTrafficPolicies []*egv
 			res = append(res, policy)
 
 			// Negative statuses have already been assigned so its safe to skip
-			gateway := resolveBTPolicyGatewayTargetRef(policy, gatewayMap)
+			gateway := resolveBTPolicyGatewayTargetRef(policy, t.GatewayControllerName, gatewayMap)
```

Comment:
To simplify the code structure, I would like return errors directly from sub-methods and consolidate status handling within the outermost layer. This approach would also improve readability. 

It's just my preferred coding flavor, I recognize others may have different approaches. 

This is not blocking, we can handle it later if it makes sense.

Reply:
make sense, I have done some refactor.